test/metabase/api/user_test.clj

```diff
@@ -60,34 +60,36 @@
 
 (deftest user-list-test
   (testing "GET /api/user"
-    (testing "Check that anyone can get a list of all active Users"
+    (testing "Check that admins can get a list of all active Users"
       (mt/with-non-admin-groups-no-root-collection-perms
-        (is (= [{:id          (mt/user->id :crowberto)
-                 :email       "crowberto@metabase.com"
-                 :first_name  "Crowberto"
-                 :last_name   "Corv"
-                 :common_name "Crowberto Corv"}
-                {:id          (mt/user->id :lucky)
-                 :email       "lucky@metabase.com"
-                 :first_name  "Lucky"
-                 :last_name   "Pigeon"
-                 :common_name "Lucky Pigeon"}
-                {:id          (mt/user->id :rasta)
-                 :email       "rasta@metabase.com"
-                 :first_name  "Rasta"
-                 :last_name   "Toucan"
-                 :common_name "Rasta Toucan"}]
-               (->> ((mt/user-http-request :rasta :get 200 "user") :data)
-                    (filter mt/test-user?))))))
-    (testing "Get list of users with a query"
+        (is (= #{"crowberto@metabase.com"
+                 "lucky@metabase.com"
+                 "rasta@metabase.com"}
+               (->> ((mt/user-http-request :crowberto :get 200 "user") :data)
+                    (filter mt/test-user?)
+                    (map :email)
+                    set)))
+        (testing "with a query"
+          (is (= "lucky@metabase.com"
+                 (-> (mt/user-http-request :crowberto :get 200 "user" :query "lUck") :data first :email))))))
+    (testing "Check that non-admins cannot get a list of all active Users"
       (mt/with-non-admin-groups-no-root-collection-perms
-        (is (= [{:id          (mt/user->id :lucky)
-                 :email       "lucky@metabase.com"
-                 :first_name  "Lucky"
-                 :last_name   "Pigeon"
-                 :common_name "Lucky Pigeon"}]
-               (->> ((mt/user-http-request :rasta :get 200 "user" :query "lUck") :data)
-                    (filter mt/test-user?))))))))
+        (is (= "You don't have permissions to do that."
+               (mt/user-http-request :lucky :get 403 "user")))))
+    (testing "Check that group-managers can get a list of that group's users."
+      (premium-features-test/with-premium-features #{:advanced-permissions}
+        (t2.with-temp/with-temp
+            [:model/PermissionsGroup           {group-id1 :id} {:name "Test Group"}
+             :model/PermissionsGroupMembership _ {:user_id (mt/user->id :rasta) :group_id group-id1 :is_group_manager true}
+             :model/PermissionsGroupMembership _ {:user_id (mt/user->id :lucky) :group_id group-id1 :is_group_manager false}]
```

Comment:
Two more test cases, pursuant to my comment above:
* A manager of one group can't list users that are in a different group which they are a member, but not a manager
* A manager of one group can't list users in groups which they are not a member

Reply:
added tests to cover these cases, thanks for pointing them out :)